### PR TITLE
Google map component

### PR DESCRIPTION
**Pull request by RumataDeEstor**
This PR adds GoogleMap Component which is powered by Google API script added to main page. 
screenshot 2017-08-08 15-34-33

### CODE REVIEW

File: components/googleMap/googleMap.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
+import React, { Component } from 'react';
+import { PropTypes } from 'prop-types';
```
Comment (fahad19) [inline components/googleMap/googleMap.js]: I wasn't aware PropTypes can be imported like this (destructured) from `prop-types` package too.

Comment (RumataDeEstor) [inline components/googleMap/googleMap.js]: *unified*

Comment (fahad19) [inline components/googleMap/googleMap.js]: both seem to work 👍 

tried out here: https://codesandbox.io/s/KrmEjrmpz

would be nice if the project sticks to one way of using it though across all the components.

File: components/googleMap/googleMap.js [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+      this.latlng = position;
+      this.info = info;
+      this.index = index;
+      this.setMap(map);
+    }
+
+    CustomMarker.prototype = new google.maps.OverlayView();
+
+    CustomMarker.prototype.draw = function draw() {
+      let div = this.div;
```
Comment (asci) [inline components/googleMap/googleMap.js]: Is it possible to move this outside the `GoogleMap` component class?  having different `this` is really confusing

File: components/googleMap/googleMap.js [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    function CustomMarker(pointData, map, index) {
+      const { position, info } = pointData;
+
+      this.latlng = position;
+      this.info = info;
+      this.index = index;
+      this.setMap(map);
+    }
+
+    CustomMarker.prototype = new google.maps.OverlayView();
```
Comment (asci) [inline components/googleMap/googleMap.js]: any checks that `google` is defined?

File: components/googleMap/googleMap.js [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+      let div = this.div;
+      if (!div) {
+        div = document.createElement('DIV');
+        this.div = div;
+        div.style.position = 'absolute';
+        div.style.paddingLeft = '0px';
+        div.style.cursor = 'pointer';
+        div.style.width = '10px';
+        div.style.height = '10px';
+        div.classList.add('ui-google-map__custom-marker');
```
Comment (asci) [inline components/googleMap/googleMap.js]: is it possible to move all `div.style` under this class rule in CSS?

File: components/googleMap/googleMap.js (hunk 1 above)
Comment (RumataDeEstor) [inline components/googleMap/googleMap.js]: @fahad19 yep, I fixed it. Let's keep using as a default export :)
but interesting to know :)

File: components/googleMap/googleMap.js [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    });
+  }
+
+  /**
+   * Sets up a map with passed options and calls a corresponding methods for creating markers
+   *
+   * @method linkRef
+   * @param {Object} ref - node object to be linked as a ref; a map will be rendered here
+   */
+  linkRef = (ref) => {
```
Comment (asci) [inline components/googleMap/googleMap.js]: why arrow function?

File: components/googleMap/googleMap.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      enableTestCustomMarker: false,
+      points: [{
+        position: { lat: 52.379189, lng: 4.899431 },
+        customMarker: true,
+        info: {
+          title: 'custom title',
+        },
+      }],
+      center: { lat: 52.379189, lng: 4.899431 },
+      testCustomStyles: [
```
Comment (asci) [inline components/googleMap/googleMap.md]: why it is called `__test__CustomStyles`?

File: components/googleMap/googleMap.scss [hunk 7]
```diff
@@ -0,0 +1,26 @@
+.ui-google-map__custom-marker {
```
Comment (asci) [inline components/googleMap/googleMap.scss]: could be just `css` file?

File: components/googleMap/googleMap.js (hunk 5 above)
Comment (RumataDeEstor) [inline components/googleMap/googleMap.js]: @asci as a short alternative to `bind`

File: components/googleMap/googleMap.md (hunk 6 above)
Comment (RumataDeEstor) [inline components/googleMap/googleMap.md]: @asci I can't get the problem with that

File: components/googleMap/googleMap.scss (hunk 7 above)
Comment (RumataDeEstor) [inline components/googleMap/googleMap.scss]: Why? As I see atm we have `.scss` everywhere

Comment (asci) [inline components/googleMap/googleMap.scss]: not everywhere https://github.com/Travix-International/travix-ui-kit/blob/7f5d9bfdb31a0cae9550a2d4a370a5f170d42a49/components/index.scss#L6

we're possible will move out of SCSS, so if you don't use any significant features — why not to start use plain CSS just now?

File: components/googleMap/googleMap.md (hunk 6 above)
Comment (asci) [inline components/googleMap/googleMap.md]: just wondering word `test` in props. Was it made for testing only?

Comment (RumataDeEstor) [inline components/googleMap/googleMap.md]: @asci I tried to mark that it is only example, not the exact implementation of marker / styles. But np, I removed word `test`, just put '(example)' there.

File: components/googleMap/googleMap.js (hunk 5 above)
Comment (asci) [inline components/googleMap/googleMap.js]: Oh, ok, got it, thanks! Wasn't aware that is it already on stage 2

File: components/googleMap/googleMap.js (hunk 3 above)
Comment (AlleeX) [inline components/googleMap/googleMap.js]: @asci done. Thanks!

File: components/googleMap/googleMap.js (hunk 2 above)
Comment (AlleeX) [inline components/googleMap/googleMap.js]: @asci done. Thanks! ;)

File: components/googleMap/googleMap.js (hunk 4 above)
Comment (AlleeX) [inline components/googleMap/googleMap.js]: @asci done. Thanks!

File: components/googleMap/googleMap.scss (hunk 7 above)
Comment (AlleeX) [inline components/googleMap/googleMap.scss]: @asci Because in this file we are use '&'

Comment (asci) [inline components/googleMap/googleMap.scss]: yes, but you can easily avoid using it and this will make file even better (with pure flat structure)

Comment (AlleeX) [inline components/googleMap/googleMap.scss]: `.ui-google-map__custom-marker {
...
}

.ui-google-map__custom-marker:before {
...
}

.ui-google-map__custom-marker:after {
...
}`

@asci Do you mean Is it more pure flat structure (copypaste class names) ??

Then why did we start using sass before? :)

**Comment by AlleeX** [comment]
@asci all your remarks has been fixed. Thanks a for for review!)

File: components/googleMap/googleMap.scss (hunk 7 above)
Comment (asci) [inline components/googleMap/googleMap.scss]: because we used SASS variables, now we're moving to CSS Variables

**Comment by AlleeX** [comment]
@mAiNiNfEcTiOn regarding sass... but in the PR, which was merged today, guys used sass. Strange... but ok. I'll change

**Comment by AlleeX** [comment]
@asci @mAiNiNfEcTiOn converting .scss to .css has been done

**Comment by asci** [comment]
regarding the PR, which was merged today — I didn't notice, my bad.

**Comment by mAiNiNfEcTiOn** [comment]
@AlleeX indeed, must've missed it. Not personal, though 😉

**Comment by mAiNiNfEcTiOn** [comment]
@asci Changes requested were done, I'm merging.

**Pull request merged**